Automattic/wp-calypso
# Jetpack Backup: Prevent restores with invalid creds

We require to prevent site restores if Jetpack Credentials are no longer valid.
We implemented a way to validate credentials in #66468 and #67589 on Calypso Blue and Calypso Green settings page. The idea is to reuse it and disable restore buttons as soon as we detect they are invalid.

#### Proposed changes
- [x] Disable `Restore to this point` button in actions toolbar on Activity Log when credentials are invalid
- [x] Disable `Restore to this point` button in main backups page when credentials are invalid
- [x] Show banner requesting credentials if credentials are invalid
- [x] Introduce new `QueryJetpackCredentialsStatus` data component that will trigger actions for testing Jetpack Credentials to determinate whether they are still valid or not.
- [x] Trigger credentials test validation in backup page
- [x] Trigger credentials test validation in Activity List component

#### Reducer changes
- [x] Jetpack Credentials `testRequestStatus` reducer is now used for handling the test credentials request status, so we can use it to validate is there is an ongoing request to avoid unnecessary API calls.
- [x] Jetpack Credentials `testStatus` reducer is now used for handling the status of the credentials. It will handle `pending`, `valid` and `invalid` statuses.
- [x] Refactored `getJetpackCredentialsTestStatus` and unit tests to consider the changes in reducers.

#### Screenshots

We are reusing the same messages and behaviors that we have when the customer has no credentials.

Component | Valid credentials | Invalid credentials
--- | --- | ---
Backup page (Banner and Daily Backup) | ![image](https://user-images.githubusercontent.com/1488641/194476053-90efa5eb-d2eb-4e4c-8586-9c141945abc0.png) | ![image](https://user-images.githubusercontent.com/1488641/194476175-4e1914b3-7839-47ca-8a4c-93a5602ebd28.png)
Actions toolbar in Activity Log item (Backup and Activity Log pages) | ![image](https://user-images.githubusercontent.com/1488641/194476263-e8f0a6a8-fd81-47d9-89d4-4588c4c57e57.png) |  ![image](https://user-images.githubusercontent.com/1488641/194476326-e06ae093-c02d-4500-b418-74fb58afde4b.png)

#### Testing Instructions


* **Jetpack Cloud / Calypso Green**
  * Grab a site with a Jetpack Backup plan. Preferable, a site you can modify the SSH/SFTP password.
  * Ensure you have working SSH/SFTP credentials on `Settings` page.
  * Navigate to `Activity Log` and `Backup` sections and ensure you can see the `Restore to this point` buttons in green color (that means they are enabled).
    * Also you should see the banner with the message _“Every change you make will be backed up”_ on top of Backup page.
  * Modify the SSH/SFTP password on your site server (not on Jetpack Cloud settings).
  * Navigate again to `Activity Log` or `Backup` page and validate the same buttons are now disabled. It may take a couple of seconds as it is trying to test credentials using an API.

#### Pre-merge Checklist






#### References
* peaFOp-7G-p2
* 1202895132770065-as-1202906937479000
